superset/connectors/sqla/models.py

```diff
@@ -879,15 +892,17 @@ def adhoc_metric_to_sqla(
         label = utils.get_metric_name(metric)
 
         if expression_type == utils.AdhocMetricExpressionType.SIMPLE:
-            column_name = metric["column"].get("column_name")
+            column_name = cast(str, metric["column"].get("column_name"))
```

Comment:
This cast is necessary, as it's not possible to do proper union types where `column_name` is always defined if `expression_type == 'SIMPLE'` but `None` if `expression_type == 'SQL'`